pkg/types/100/types.go

```diff
@@ -92,7 +92,7 @@ type Result struct {
 	Interfaces []*Interface   `json:"interfaces,omitempty"`
 	IPs        []*IPConfig    `json:"ips,omitempty"`
 	Routes     []*types.Route `json:"routes,omitempty"`
-	DNS        types.DNS      `json:"dns,omitempty"`
+	DNS        *types.DNS     `json:"dns,omitempty"`
```

Comment:
Unfortunately, we can't change the API here, since it would break upgrades :-(.

Reply:
OK, I will change the way to keep the structure type in another PRs. thank you for comment!

Reply:
@squeed create another PR for alternative: https://github.com/containernetworking/cni/pull/1039 PTAL?